src/main/java/org/example/exceptions/InvalidRiderIDException.java

```diff
@@ -0,0 +1,7 @@
+package org.example.exceptions;
+
+public class InvalidRiderIDException extends RuntimeException {
+    public InvalidRiderIDException() {
```

Comment:
Take error message and throwable object as the parameters in this constructor to capture the relevant details. Read more about exception chaining in java. 
https://www.baeldung.com/java-chained-exceptions

Do the same for all custom exceptions.